api/scanner/scanner_media.go

```diff
@@ -8,24 +8,92 @@ import (
 	"log"
 	"os"
 	"path"
+	"path/filepath"
+	"strings"
 
 	"github.com/pkg/errors"
 	"github.com/viktorstrate/photoview/api/graphql/models"
 )
 
-func scanForSideCarFile(path string) *string {
-	testPath := path + ".xmp"
+func isFileExists(testPath string) bool {
 	_, err := os.Stat(testPath)
 
 	if os.IsNotExist(err) {
-		return nil
+		return false
 	} else if err != nil {
 		// unexpected error logging
 		log.Printf("ERROR: %s", err)
-		return nil
+		return false
+	}
+	return true
+}
+
+func scanForSideCarFile(path string) *string {
+	testPath := path + ".xmp"
+
+	if isFileExists(testPath) {
+		return &testPath
+	}
+
+	return nil
+}
+
+func scanForRawCounterpartFile(imagePath string) *string {
+	ext := filepath.Ext(imagePath)
+	fileExtType, found := fileExtensions[strings.ToLower(ext)]
+
+	if found {
+		if !fileExtType.isBasicTypeSupported() {
+			return nil
+		}
+	}
+
+	pathWithoutExt := strings.TrimSuffix(imagePath, path.Ext(imagePath))
+	for ext, mediaType := range fileExtensions {
+		for _, rawType := range RawMimeTypes {
+			if rawType == mediaType {
+				testPathLowercase := pathWithoutExt + ext
+				if isFileExists(testPathLowercase) {
+					return &testPathLowercase
+				}
+
+				testPathUppercase := pathWithoutExt + strings.ToUpper(ext)
+				if isFileExists(testPathUppercase) {
+					return &testPathUppercase
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func scanForCompressedCounterpartFile(imagePath string) *string {
+	ext := filepath.Ext(imagePath)
+	fileExtType, found := fileExtensions[strings.ToLower(ext)]
+
+	if found {
+		if fileExtType.isBasicTypeSupported() {
+			return nil
+		}
+	}
+
+	pathWithoutExt := strings.TrimSuffix(imagePath, path.Ext(imagePath))
+	for ext, mediaType := range fileExtensions {
```

Comment:
Maybe add a helper method for `MediaType`s to get the corresponding extensions, something like this

```go
func (mediaType *MediaType) FileExtensions() []string { ... }
```

Then we could just loop over the extension like so

```go
for extension := range TypeJpeg.FileExtensions() { ... }
```

Reply:
Thanks for pointing me this out, it's really neat idea!

Reply:
I used `mediaType MediaType` instead of `mediaType *MediaType` in order to use it directly with constants such `TypeJpeg` otherwise I am getting an error TypeRange is not a pointer.